addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterAddOn is the Custom Resource object which holds the current state
+// of an add-on. This object is used by add-on operators to convey their state.
+// The resource should be created in the ManagedCluster's cluster namespace.
+type ManagedClusterAddOn struct {
```

Comment:
Try to stay away from possessive and rewrite:

Create the resource in the namespace for the ManagedCluster.